### PR TITLE
Field error border addition

### PR DESCRIPTION
**Pull request by abbasali1994**
Ticket(s):
- [Add red border to fields that have an error attached to them#1929](https://github.com/AngelProtocolFinance/angelprotocol-web-app/issues/1929)

## Instructions on making this work

- run `yarn` or `yarn install` to install npm dependencies
- run `yarn run test --watchAll` to verify all tests still pass
- (optional) run `yarn run build` to verify the build passes
- run `yarn start` to start the webapp
-

## UI changes for review

When major UI changes are introduced with a PR, please include links to URLS to compare or screenshots demonstrating the difference and notify on design changes

### CODE REVIEW

**Comment by abbasali1994** [comment]
Please mention any field components that i might have missed which require these borders to be included

File: src/components/CountrySelector/index.tsx [hunk 1]
```diff
@@ -68,7 +69,7 @@ export default function CountrySelector<
       as="div"
       className={`relative items-center grid grid-cols-[auto_auto_1fr] w-full field-container ${
         props.classes?.container || ""
-      }`}
+      } ${getErrorClasses(errors, `${props.fieldName}.name`)}`}
```
Comment (ap-justin) [inline src/components/CountrySelector/index.tsx:72]: @abbasali1994
Another way to attach these error classes would be to add aria attribute

```ts
const isError = !!errors[name /** usually provided*/]?.message
<Combobox aria-invalid={isError}/>
```
and in `index.css` could put classes in `field-container` and `field-input` utilities
```
@apply ... aria-invalid:border-red dark:aria-invalid:border-red

```
this way no need to generalize gettng error info as the nesting depends on every component as shown in this function

```
export const resolvePath = (object: any, path: string) =>
  path.split(".").reduce((o, p) => (o ? o[p] : {}), object);
```

**Review by ap-justin — CHANGES_REQUESTED** [review]

File: src/components/form/Field.tsx [hunk 2]
```diff
@@ -53,7 +55,7 @@ export function Field<T extends FieldValues, K extends InputType = "text">({
         ...(type === textarea ? {} : { type }),
         id,
         disabled: isSubmitting || disabled,
-        className: input,
+        className: `${input} ${getErrorClasses(errors, name)}`,
```
Comment (ap-justin) [inline src/components/form/Field.tsx:58]: @abbasali1994, @SovereignAndrey

I think these red borders should also be applied to other form UIs for consistency 
image

Comment (SovereignAndrey) [reply src/components/form/Field.tsx:58]: > I think these red borders should also be applied to other form UIs (rich text, selectors, dropzones) for consistency image

@ap-justin, agreed that's best for consistency of UI. 
@abbasali1994, please add that into this PR. Thanks! :pray:

Comment (abbasali1994) [reply src/components/form/Field.tsx:58]: okay

Comment (abbasali1994) [reply src/components/form/Field.tsx]: Added error borders for multiselects and dropzones

Comment (abbasali1994) [reply src/components/form/Field.tsx]: Implemented logic using aria-invalid like suggested by @ap-justin

File: src/components/CountrySelector/index.tsx (hunk 1 above)
Comment (abbasali1994) [reply src/components/CountrySelector/index.tsx]: Done

File: src/helpers/resolveJsonPath.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export const resolvePath = (object: any, path: string) =>
```
Comment (ap-justin) [inline src/helpers/resolveJsonPath.ts:1]: @abbasali1994, no need for this helper found in `react-hook-form/ErrorMessage` you could use [`get`](https://github.com/react-hook-form/error-message/blob/master/src/ErrorMessage.tsx#L21) util

File: package.json [hunk 4]
```diff
@@ -120,6 +120,7 @@
     "react-test-renderer": "^18.2.0",
     "source-map-explorer": "^2.5.2",
     "tailwindcss": "^3.2.7",
+    "tailwindcss-aria-attributes": "^1.3.0",
```
Comment (ap-justin) [inline package.json:123]: @abbasali1994, `aria-*` states are already included in `3.2.7`. I think there's no need for plugin for this - As other classes like `aria-disabled:*` or `aria-selected:*` is working

**Review by ap-justin — CHANGES_REQUESTED** [review]

File: src/helpers/resolveJsonPath.ts (hunk 3 above)
Comment (abbasali1994) [reply src/helpers/resolveJsonPath.ts]: DOne

File: package.json (hunk 4 above)
Comment (abbasali1994) [reply package.json]: aria-invalid isnt working as part of class def without the use of dependency (gives undefined error), but when defined seperately it works for tailwindcss, so modified index.css to remove the additional dependency

File: yarn.lock [hunk 5]
```diff
@@ -6,13 +6,13 @@ __metadata:
   cacheKey: 8
 
 "@adobe/css-tools@npm:^4.0.1":
-  version: 4.0.1
-  resolution: "@adobe/css-tools@npm:4.0.1"
-  checksum: 80226e2229024c21da9ffa6b5cd4a34b931f071e06f45aba4777ade071d7a6c94605cf73b13718b0c4b34e8b124c65c607b82eaa53a326d3eb73d9682a04a593
+  version: 4.1.0
```
Comment (ap-justin) [inline yarn.lock:9]: @abbasali1994, why is `yarn.lock` changed whilst no changes in `package.json` - did you by any change deleted this file and reinstalled deps again ?

**Comment by SovereignAndrey** [comment]
@abbasali1994 please double check all changes as I cannot get past installing now to run the branch and check thing. Thanks!

File: yarn.lock (hunk 5 above)
Comment (abbasali1994) [reply yarn.lock:9]: i will check , i just ran yarn install instead of npm install , and removed the additional dependency like you requested

Comment (abbasali1994) [reply yarn.lock:9]: yep i checked , if i revert it back i am getting lint errors, running yarn install brings all these changes to the file

File: yarn.lock [hunk 6]
```diff
@@ -6,13 +6,13 @@ __metadata:
   cacheKey: 8
 
 "@adobe/css-tools@npm:^4.0.1":
-  version: 4.0.1
```
Comment (ap-justin) [inline yarn.lock:9]: should not have changes in `yarn.lock` without changes in dependency `package.json`
see #2034

**Review by ap-justin — CHANGES_REQUESTED** [review]

File: yarn.lock (hunk 5 above)
Comment (abbasali1994) [reply yarn.lock:9]: justin I am not contradicting that it won't work, it's not gonna work for me cause of the other PR I ended up with a lot of build issues and so I had to delete yarn.lock, and eventually clean clone everything, if your PR is an exact copy of mine lets just use that and get this code merged, cause no matter what i do on my local unless i delete the lock file and do a yarn install again my code doesn't work

File: yarn.lock (hunk 6 above)
Comment (abbasali1994) [reply yarn.lock:9]: ok pls set this PR as ready for review and lets get that reviewed and merged

**Review by ap-justin — APPROVED** [review]

**Pull request merged**